pkg/gameservers/controller.go

```diff
@@ -646,7 +646,7 @@ func (c *Controller) sidecar(gs *agonesv1.GameServer) corev1.Container {
 	if !c.sidecarCPULimit.IsZero() {
 		limits[corev1.ResourceCPU] = c.sidecarCPULimit
 	}
-	if !c.sidecarCPULimit.IsZero() {
+	if !c.sidecarMemoryLimit.IsZero() {
 		limits[corev1.ResourceMemory] = c.sidecarMemoryLimit
```

Comment:
Since we're here, do we want to put in some `logrus.Warnf()` log messages for both of these -- otherwise they fail silently.

WDYT?

Reply:
Will update this

Reply:
What Warn message should I add?
Similar to this one:
QoS would not be assigned to "Guaranteed" unless CPU and Memory are the same in Limit and Request for the sidecar?

Reply:
Something like:
```
	if !c.sidecarMemoryLimit.IsZero() {
		limits[corev1.ResourceMemory] = c.sidecarMemoryLimit
       } else {
             c.logger.Warnf("Sidecar memory limit of %s is less than or equal to zero. Skipping.", c.sidecarMemoryLimit)
       }
```

I was thinking something more like that? 

I feel like if people are changing memory and cpu limits, then they know what they are doing, and it's up to them to control it as they see fit. But a 0 or negative value isn't going to work, so we should warn people that we are skipping setting the value.

Sound good?

Reply:
Mark thanks for this request, I have added validation on the place where we set those parameters, not where we use them.
So we will close #1419 issue.